### PR TITLE
support configuration for keepalive parameters

### PR DESCRIPTION
**Pull request by cch123**
Let the heartbeat support configuration, we can tune the parameters to find out the best values for heartbeat

### CODE REVIEW

**Review by nejisama — APPROVED** [review]

**Comment by cch123** [comment]
image

**Comment by taoyuanyuan** [comment]
add testcase for new scene.

**Comment by taoyuanyuan** [comment]
add beachmark for uber atomic and runtime atomic.

File: pkg/stream/xprotocol/keepalive.go [hunk 1]
```diff
@@ -121,14 +147,20 @@ func (kp *xprotocolKeepAlive) StartIdleTimeout() {
...
 	id := sender.GetStream().ID()
+
 	// check idle free
 	if kp.idleFree.CheckFree(id) {
 		kp.Codec.Close()
 		return
 	}
+
+	// reset the tick count
+	kp.tickCount.Store(0)
```
Comment (taoyuanyuan) [inline pkg/stream/xprotocol/keepalive.go:162]: ResetTick()

**Comment by taoyuanyuan** [comment]
这样处理还是有问题的，如果一直有请求，ResetTick一直清空tick，导致不会发送心跳包，假死连接将断不掉了

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Pull request merged**